tests/scripts/basic-build-test.sh

```diff
@@ -104,7 +104,14 @@ echo
 
 # Step 3 - Process the coverage report
 cd ..
-make lcov |tee tests/cov-$TEST_OUTPUT
+{
+    make lcov
+    echo SUCCESS
+} | tee tests/cov-$TEST_OUTPUT
+
+if [ "$(tail -n1 tests/cov-$TEST_OUTPUT)" != "SUCCESS" ]; then
+    echo >&2 "Fatal: 'make lcov' failed"
```

Comment:
The commit message says "exit immediately" but I'm not seeing what part of the code actually does that.

Reply:
Oops. I amended the commit.